include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -529,19 +638,28 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Swizzle,Dither,Nor
 								for (auto h=0; h<MaxChannels; h++)
 									value[h] += windowSample[h];
 							};
-							// do the filtering 
+							// do the filtering
 							core::vectorSIMDf tmp;
 							tmp[axis] = float(i)+0.5f;
 							core::vectorSIMDi32 windowCoord(0);
 							windowCoord[axis] = kernel.getWindowMinCoord(tmp*fScale,tmp)[axis];
 							auto relativePos = tmp[axis]-float(windowCoord[axis]);
+							const uint32_t phaseIndex = i % phaseCount[axis];
 							for (auto h=0; h<windowSize; h++)
 							{
-								value_type windowSample[MaxChannels];
+								if (state->enableLUTUsage)
+								{
+									for (auto ch = 0; ch < MaxChannels; ch++)
+										value[ch] += phaseSupportLUTPixel[phaseIndex * windowSize + h] * lineBuffer[(windowCoord[axis] - windowMinCoord[axis]) * MaxChannels + ch];
```

Comment:
1) ok I can see that as you move positively in the LUT, you also move positively in the window coord.

However convolution moves in reverse in the second function (as you can see with `relativePos -= 1.f`), so is your LUT inverted then, and did you test and validate the blit still works with a non-symmetric Kernel such as one used for derivative maps?

Reply:
2. I'd turn `phaseSupportLUTPixel[phaseIndex * windowSize + h] * lineBuffer[(windowCoord[axis] - windowMinCoord[axis]) * MaxChannels + ch]` into a lambda

Reply:
Resolved in #343.